pkg/operator/cassandra/sidecar/config/cassandra.go

```diff
@@ -0,0 +1,316 @@
+package config
+
+// Configuration represents the content of cassandra.yaml file
+// see https://cassandra.apache.org/doc/latest/configuration/cassandra_config_file.html for full explanations
+// This struct was generated from a cassandra 3.11 config file
+type Configuration struct {
+	ClusterName                string `json:"cluster_name"`
+	NumTokens                  int    `json:"num_tokens"`
+	HintedHandoffEnabled       bool   `json:"hinted_handoff_enabled"`
+	MaxHintWindowInMs          int    `json:"max_hint_window_in_ms"`
+	HintedHandoffThrottleInKb  int    `json:"hinted_handoff_throttle_in_kb"`
+	MaxHintsDeliveryThreads    int    `json:"max_hints_delivery_threads"`
+	HintsFlushPeriodInMs       int    `json:"hints_flush_period_in_ms"`
+	MaxHintsFileSizeInMb       int    `json:"max_hints_file_size_in_mb"`
+	BatchlogReplayThrottleInKb int    `json:"batchlog_replay_throttle_in_kb"`
+	Authenticator              string `json:"authenticator"`
+	Authorizer                 string `json:"authorizer"`
+	RoleManager                string `json:"role_manager"`
+	RolesValidityInMs          int    `json:"roles_validity_in_ms"`
+	PermissionsValidityInMs    int    `json:"permissions_validity_in_ms"`
+	CredentialsValidityInMs    int    `json:"credentials_validity_in_ms"`
+	Partitioner                string `json:"partitioner"`
+	CdcEnabled                 bool   `json:"cdc_enabled"`
+	DiskFailurePolicy          string `json:"disk_failure_policy"`
+	CommitFailurePolicy        string `json:"commit_failure_policy"`
+	// TODO
+	//PreparedStatementsCacheSizeMb       interface{} `json:"prepared_statements_cache_size_mb"`
+	//ThriftPreparedStatementsCacheSizeMb interface{} `json:"thrift_prepared_statements_cache_size_mb"`
+	//KeyCacheSizeInMb   interface{} `json:"key_cache_size_in_mb"`
```

Comment:
Are these and the other commented out options going to be implemented or removed from the code after review?

Reply:
The idea was to leave this as a TODO in the code - if someone needs a particular config field one can use the search to find it in the code. I don't know yet how to handle the types for these configuration values (docs say: `Default value is empty to make it "auto"`).

I can remove it or leave a comment with more details if you want

Reply:
I wonder if `0` would be "empty" for Cassandra in that case, then we could have them in there as integers or can it also be a string (e.g., `128M` or similar).

If `0` doesn't mean they are "empty", then let's change the comment to exactly that information, that they are not included to the need of being empty, on the other hand Go's json tags should also allow for `omitempty` which might do the trick here. Could you check that if it is possible?

Reply:
I did research on that: In some cases`0` means disable cache key size:

```yaml
# Maximum size of the key cache in memory.
#
# Each key cache hit saves 1 seek and each row cache hit saves 2 seeks at the
# minimum, sometimes more. The key cache is fairly tiny for the amount of
# time it saves, so it's worthwhile to use it at large numbers.
# The row cache saves even more time, but must contain the entire row,
# so it is extremely space-intensive. It's best to only use the
# row cache if you have hot rows or static rows.
#
# NOTE: if you reduce the size, you may not get you hottest keys loaded on startup.
#
# Default value is empty to make it "auto" (min(5% of Heap (in MB), 100MB)). Set to 0 to disable key cache.
key_cache_size_in_mb:
```
source: taken from cassandra:3.11 image `/etc/cassandra/cassandra.yaml`

`omitempty` does the trick. cassandra does not complain, thank you!